// vue-global-config/dist/vue-global-config.mjs
import { isVue3 as Dt } from "vue-demi";
var Nt = typeof global == "object" && global && global.Object === Object && global;
const it = Nt;
var Gt = typeof self == "object" && self && self.Object === Object && self, Ut = it || Gt || Function("return this")();
const m = Ut;
var Bt = m.Symbol;
const O = Bt;
var ot = Object.prototype, Ht = ot.hasOwnProperty, Kt = ot.toString, G = O ? O.toStringTag : void 0;
function Wt(e) {
  var t = Ht.call(e, G), r = e[G];
  try {
    e[G] = void 0;
    var n = !0;
  } catch {
  }
  var a = Kt.call(e);
  return n && (t ? e[G] = r : delete e[G]), a;
}
var zt = Object.prototype, Zt = zt.toString;
function Xt(e) {
  return Zt.call(e);
}
var qt = "[object Null]", Yt = "[object Undefined]", Ie = O ? O.toStringTag : void 0;
function j(e) {
  return e == null ? e === void 0 ? Yt : qt : Ie && Ie in Object(e) ? Wt(e) : Xt(e);
}
function w(e) {
  return e != null && typeof e == "object";
}
var Jt = "[object Symbol]";
function ye(e) {
  return typeof e == "symbol" || w(e) && j(e) == Jt;
}
function Qt(e, t) {
  for (var r = -1, n = e == null ? 0 : e.length, a = Array(n); ++r < n; )
    a[r] = t(e[r], r, e);
  return a;
}
var Vt = Array.isArray;
const $ = Vt;
var kt = 1 / 0, je = O ? O.prototype : void 0, xe = je ? je.toString : void 0;
function st(e) {
  if (typeof e == "string")
    return e;
  if ($(e))
    return Qt(e, st) + "";
  if (ye(e))
    return xe ? xe.call(e) : "";
  var t = e + "";
  return t == "0" && 1 / e == -kt ? "-0" : t;
}
function P(e) {
  var t = typeof e;
  return e != null && (t == "object" || t == "function");
}
function be(e) {
  return e;
}
var er = "[object AsyncFunction]", tr = "[object Function]", rr = "[object GeneratorFunction]", nr = "[object Proxy]";
function ve(e) {
  if (!P(e))
    return !1;
  var t = j(e);
  return t == tr || t == rr || t == er || t == nr;
}
var ar = m["__core-js_shared__"];
const oe = ar;
var Me = function() {
  var e = /[^.]+$/.exec(oe && oe.keys && oe.keys.IE_PROTO || "");
  return e ? "Symbol(src)_1." + e : "";
}();
function ir(e) {
  return !!Me && Me in e;
}
var or = Function.prototype, sr = or.toString;
function x(e) {
  if (e != null) {
    try {
      return sr.call(e);
    } catch {
    }
    try {
      return e + "";
    } catch {
    }
  }
  return "";
}
var fr = /[\\^$.*+?()[\]{}|]/g, ur = /^\[object .+?Constructor\]$/, cr = Function.prototype, lr = Object.prototype, pr = cr.toString, gr = lr.hasOwnProperty, dr = RegExp(
  "^" + pr.call(gr).replace(fr, "\\$&").replace(/hasOwnProperty|(function).*?(?=\\\()| for .+?(?=\\\])/g, "$1.*?") + "$"
);
function hr(e) {
  if (!P(e) || ir(e))
    return !1;
  var t = ve(e) ? dr : ur;
  return t.test(x(e));
}
function yr(e, t) {
  return e == null ? void 0 : e[t];
}
function M(e, t) {
  var r = yr(e, t);
  return hr(r) ? r : void 0;
}
var br = M(m, "WeakMap");
const ue = br;
var Le = Object.create, vr = function() {
  function e() {
  }
  return function(t) {
    if (!P(t))
      return {};
    if (Le)
      return Le(t);
    e.prototype = t;
    var r = new e();
    return e.prototype = void 0, r;
  };
}();
const _r = vr;
function Tr(e, t, r) {
  switch (r.length) {
    case 0:
      return e.call(t);
    case 1:
      return e.call(t, r[0]);
    case 2:
      return e.call(t, r[0], r[1]);
    case 3:
      return e.call(t, r[0], r[1], r[2]);
  }
  return e.apply(t, r);
}
function ft(e, t) {
  var r = -1, n = e.length;
  for (t || (t = Array(n)); ++r < n; )
    t[r] = e[r];
  return t;
}
var $r = 800, Ar = 16, Or = Date.now;
function wr(e) {
  var t = 0, r = 0;
  return function() {
    var n = Or(), a = Ar - (n - r);
    if (r = n, a > 0) {
      if (++t >= $r)
        return arguments[0];
    } else
      t = 0;
    return e.apply(void 0, arguments);
  };
}
function mr(e) {
  return function() {
    return e;
  };
}
var Pr = function() {
  try {
    var e = M(Object, "defineProperty");
    return e({}, "", {}), e;
  } catch {
  }
}();
const J = Pr;
var Sr = J ? function(e, t) {
  return J(e, "toString", {
    configurable: !0,
    enumerable: !1,
    value: mr(t),
    writable: !0
  });
} : be;
const Er = Sr;
var Cr = wr(Er);
const Ir = Cr;
function jr(e, t) {
  for (var r = -1, n = e == null ? 0 : e.length; ++r < n && t(e[r], r, e) !== !1; )
    ;
  return e;
}
var xr = 9007199254740991, Mr = /^(?:0|[1-9]\d*)$/;
function _e(e, t) {
  var r = typeof e;
  return t = t ?? xr, !!t && (r == "number" || r != "symbol" && Mr.test(e)) && e > -1 && e % 1 == 0 && e < t;
}
function te(e, t, r) {
  t == "__proto__" && J ? J(e, t, {
    configurable: !0,
    enumerable: !0,
    value: r,
    writable: !0
  }) : e[t] = r;
}
function z(e, t) {
  return e === t || e !== e && t !== t;
}
var Lr = Object.prototype, Fr = Lr.hasOwnProperty;
function ut(e, t, r) {
  var n = e[t];
  (!(Fr.call(e, t) && z(n, r)) || r === void 0 && !(t in e)) && te(e, t, r);
}
function R(e, t, r, n) {
  var a = !r;
  r || (r = {});
  for (var i = -1, o = t.length; ++i < o; ) {
    var s = t[i], f = n ? n(r[s], e[s], s, r, e) : void 0;
    f === void 0 && (f = e[s]), a ? te(r, s, f) : ut(r, s, f);
  }
  return r;
}
var Fe = Math.max;
function Rr(e, t, r) {
  return t = Fe(t === void 0 ? e.length - 1 : t, 0), function() {
    for (var n = arguments, a = -1, i = Fe(n.length - t, 0), o = Array(i); ++a < i; )
      o[a] = n[t + a];
    a = -1;
    for (var s = Array(t + 1); ++a < t; )
      s[a] = n[a];
    return s[t] = r(o), Tr(e, this, s);
  };
}
function Dr(e, t) {
  return Ir(Rr(e, t, be), e + "");
}
var Nr = 9007199254740991;
function Te(e) {
  return typeof e == "number" && e > -1 && e % 1 == 0 && e <= Nr;
}
function re(e) {
  return e != null && Te(e.length) && !ve(e);
}
function Gr(e, t, r) {
  if (!P(r))
    return !1;
  var n = typeof t;
  return (n == "number" ? re(r) && _e(t, r.length) : n == "string" && t in r) ? z(r[t], e) : !1;
}
function ct(e) {
  return Dr(function(t, r) {
    var n = -1, a = r.length, i = a > 1 ? r[a - 1] : void 0, o = a > 2 ? r[2] : void 0;
    for (i = e.length > 3 && typeof i == "function" ? (a--, i) : void 0, o && Gr(r[0], r[1], o) && (i = a < 3 ? void 0 : i, a = 1), t = Object(t); ++n < a; ) {
      var s = r[n];
      s && e(t, s, n, i);
    }
    return t;
  });
}
var Ur = Object.prototype;
function $e(e) {
  var t = e && e.constructor, r = typeof t == "function" && t.prototype || Ur;
  return e === r;
}
function Br(e, t) {
  for (var r = -1, n = Array(e); ++r < e; )
    n[r] = t(r);
  return n;
}
var Hr = "[object Arguments]";
function Re(e) {
  return w(e) && j(e) == Hr;
}
var lt = Object.prototype, Kr = lt.hasOwnProperty, Wr = lt.propertyIsEnumerable, zr = Re(function() {
  return arguments;
}()) ? Re : function(e) {
  return w(e) && Kr.call(e, "callee") && !Wr.call(e, "callee");
};
const Q = zr;
function Zr() {
  return !1;
}
var pt = typeof exports == "object" && exports && !exports.nodeType && exports, De = pt && typeof module == "object" && module && !module.nodeType && module, Xr = De && De.exports === pt, Ne = Xr ? m.Buffer : void 0, qr = Ne ? Ne.isBuffer : void 0, Yr = qr || Zr;
const B = Yr;
var Jr = "[object Arguments]", Qr = "[object Array]", Vr = "[object Boolean]", kr = "[object Date]", en = "[object Error]", tn = "[object Function]", rn = "[object Map]", nn = "[object Number]", an = "[object Object]", on = "[object RegExp]", sn = "[object Set]", fn = "[object String]", un = "[object WeakMap]", cn = "[object ArrayBuffer]", ln = "[object DataView]", pn = "[object Float32Array]", gn = "[object Float64Array]", dn = "[object Int8Array]", hn = "[object Int16Array]", yn = "[object Int32Array]", bn = "[object Uint8Array]", vn = "[object Uint8ClampedArray]", _n = "[object Uint16Array]", Tn = "[object Uint32Array]", h = {};
h[pn] = h[gn] = h[dn] = h[hn] = h[yn] = h[bn] = h[vn] = h[_n] = h[Tn] = !0;
h[Jr] = h[Qr] = h[cn] = h[Vr] = h[ln] = h[kr] = h[en] = h[tn] = h[rn] = h[nn] = h[an] = h[on] = h[sn] = h[fn] = h[un] = !1;
function $n(e) {
  return w(e) && Te(e.length) && !!h[j(e)];
}
function Ae(e) {
  return function(t) {
    return e(t);
  };
}
var gt = typeof exports == "object" && exports && !exports.nodeType && exports, U = gt && typeof module == "object" && module && !module.nodeType && module, An = U && U.exports === gt, se = An && it.process, On = function() {
  try {
    var e = U && U.require && U.require("util").types;
    return e || se && se.binding && se.binding("util");
  } catch {
  }
}();
const F = On;
var Ge = F && F.isTypedArray, wn = Ge ? Ae(Ge) : $n;
const Oe = wn;
var mn = Object.prototype, Pn = mn.hasOwnProperty;
function dt(e, t) {
  var r = $(e), n = !r && Q(e), a = !r && !n && B(e), i = !r && !n && !a && Oe(e), o = r || n || a || i, s = o ? Br(e.length, String) : [], f = s.length;
  for (var p in e)
    (t || Pn.call(e, p)) && !(o && // Safari 9 has enumerable `arguments.length` in strict mode.
    (p == "length" || // Node.js 0.10 has enumerable non-index properties on buffers.
    a && (p == "offset" || p == "parent") || // PhantomJS 2 has enumerable non-index properties on typed arrays.
    i && (p == "buffer" || p == "byteLength" || p == "byteOffset") || // Skip index properties.
    _e(p, f))) && s.push(p);
  return s;
}
function ht(e, t) {
  return function(r) {
    return e(t(r));
  };
}
var Sn = ht(Object.keys, Object);
const En = Sn;
var Cn = Object.prototype, In = Cn.hasOwnProperty;
function jn(e) {
  if (!$e(e))
    return En(e);
  var t = [];
  for (var r in Object(e))
    In.call(e, r) && r != "constructor" && t.push(r);
  return t;
}
function Z(e) {
  return re(e) ? dt(e) : jn(e);
}
function xn(e) {
  var t = [];
  if (e != null)
    for (var r in Object(e))
      t.push(r);
  return t;
}
var Mn = Object.prototype, Ln = Mn.hasOwnProperty;
function Fn(e) {
  if (!P(e))
    return xn(e);
  var t = $e(e), r = [];
  for (var n in e)
    n == "constructor" && (t || !Ln.call(e, n)) || r.push(n);
  return r;
}
function D(e) {
  return re(e) ? dt(e, !0) : Fn(e);
}
var Rn = ct(function(e, t, r, n) {
  R(t, D(t), e, n);
});
const Dn = Rn;
var Nn = /\.|\[(?:[^[\]]*|(["'])(?:(?!\1)[^\\]|\\.)*?\1)\]/, Gn = /^\w*$/;
function we(e, t) {
  if ($(e))
    return !1;
  var r = typeof e;
  return r == "number" || r == "symbol" || r == "boolean" || e == null || ye(e) ? !0 : Gn.test(e) || !Nn.test(e) || t != null && e in Object(t);
}
var Un = M(Object, "create");
const H = Un;
function Bn() {
  this.__data__ = H ? H(null) : {}, this.size = 0;
}
function Hn(e) {
  var t = this.has(e) && delete this.__data__[e];
  return this.size -= t ? 1 : 0, t;
}
var Kn = "__lodash_hash_undefined__", Wn = Object.prototype, zn = Wn.hasOwnProperty;
function Zn(e) {
  var t = this.__data__;
  if (H) {
    var r = t[e];
    return r === Kn ? void 0 : r;
  }
  return zn.call(t, e) ? t[e] : void 0;
}
var Xn = Object.prototype, qn = Xn.hasOwnProperty;
function Yn(e) {
  var t = this.__data__;
  return H ? t[e] !== void 0 : qn.call(t, e);
}
var Jn = "__lodash_hash_undefined__";
function Qn(e, t) {
  var r = this.__data__;
  return this.size += this.has(e) ? 0 : 1, r[e] = H && t === void 0 ? Jn : t, this;
}
function I(e) {
  var t = -1, r = e == null ? 0 : e.length;
  for (this.clear(); ++t < r; ) {
    var n = e[t];
    this.set(n[0], n[1]);
  }
}
I.prototype.clear = Bn;
I.prototype.delete = Hn;
I.prototype.get = Zn;
I.prototype.has = Yn;
I.prototype.set = Qn;
function Vn() {
  this.__data__ = [], this.size = 0;
}
function ne(e, t) {
  for (var r = e.length; r--; )
    if (z(e[r][0], t))
      return r;
  return -1;
}
var kn = Array.prototype, ea = kn.splice;
function ta(e) {
  var t = this.__data__, r = ne(t, e);
  if (r < 0)
    return !1;
  var n = t.length - 1;
  return r == n ? t.pop() : ea.call(t, r, 1), --this.size, !0;
}
function ra(e) {
  var t = this.__data__, r = ne(t, e);
  return r < 0 ? void 0 : t[r][1];
}
function na(e) {
  return ne(this.__data__, e) > -1;
}
function aa(e, t) {
  var r = this.__data__, n = ne(r, e);
  return n < 0 ? (++this.size, r.push([e, t])) : r[n][1] = t, this;
}
function S(e) {
  var t = -1, r = e == null ? 0 : e.length;
  for (this.clear(); ++t < r; ) {
    var n = e[t];
    this.set(n[0], n[1]);
  }
}
S.prototype.clear = Vn;
S.prototype.delete = ta;
S.prototype.get = ra;
S.prototype.has = na;
S.prototype.set = aa;
var ia = M(m, "Map");
const K = ia;
function oa() {
  this.size = 0, this.__data__ = {
    hash: new I(),
    map: new (K || S)(),
    string: new I()
  };
}
function sa(e) {
  var t = typeof e;
  return t == "string" || t == "number" || t == "symbol" || t == "boolean" ? e !== "__proto__" : e === null;
}
function ae(e, t) {
  var r = e.__data__;
  return sa(t) ? r[typeof t == "string" ? "string" : "hash"] : r.map;
}
function fa(e) {
  var t = ae(this, e).delete(e);
  return this.size -= t ? 1 : 0, t;
}
function ua(e) {
  return ae(this, e).get(e);
}
function ca(e) {
  return ae(this, e).has(e);
}
function la(e, t) {
  var r = ae(this, e), n = r.size;
  return r.set(e, t), this.size += r.size == n ? 0 : 1, this;
}
function E(e) {
  var t = -1, r = e == null ? 0 : e.length;
  for (this.clear(); ++t < r; ) {
    var n = e[t];
    this.set(n[0], n[1]);
  }
}
E.prototype.clear = oa;
E.prototype.delete = fa;
E.prototype.get = ua;
E.prototype.has = ca;
E.prototype.set = la;
var pa = "Expected a function";
function me(e, t) {
  if (typeof e != "function" || t != null && typeof t != "function")
    throw new TypeError(pa);
  var r = function() {
    var n = arguments, a = t ? t.apply(this, n) : n[0], i = r.cache;
    if (i.has(a))
      return i.get(a);
    var o = e.apply(this, n);
    return r.cache = i.set(a, o) || i, o;
  };
  return r.cache = new (me.Cache || E)(), r;
}
me.Cache = E;
var ga = 500;
function da(e) {
  var t = me(e, function(n) {
    return r.size === ga && r.clear(), n;
  }), r = t.cache;
  return t;
}
var ha = /[^.[\]]+|\[(?:(-?\d+(?:\.\d+)?)|(["'])((?:(?!\2)[^\\]|\\.)*?)\2)\]|(?=(?:\.|\[\])(?:\.|\[\]|$))/g, ya = /\\(\\)?/g, ba = da(function(e) {
  var t = [];
  return e.charCodeAt(0) === 46 && t.push(""), e.replace(ha, function(r, n, a, i) {
    t.push(a ? i.replace(ya, "$1") : n || r);
  }), t;
});
const va = ba;
function _a(e) {
  return e == null ? "" : st(e);
}
function yt(e, t) {
  return $(e) ? e : we(e, t) ? [e] : va(_a(e));
}
var Ta = 1 / 0;
function ie(e) {
  if (typeof e == "string" || ye(e))
    return e;
  var t = e + "";
  return t == "0" && 1 / e == -Ta ? "-0" : t;
}
function bt(e, t) {
  t = yt(t, e);
  for (var r = 0, n = t.length; e != null && r < n; )
    e = e[ie(t[r++])];
  return r && r == n ? e : void 0;
}
function $a(e, t, r) {
  var n = e == null ? void 0 : bt(e, t);
  return n === void 0 ? r : n;
}
function vt(e, t) {
  for (var r = -1, n = t.length, a = e.length; ++r < n; )
    e[a + r] = t[r];
  return e;
}
var Aa = ht(Object.getPrototypeOf, Object);
const Pe = Aa;
var Oa = "[object Object]", wa = Function.prototype, ma = Object.prototype, _t = wa.toString, Pa = ma.hasOwnProperty, Sa = _t.call(Object);
function Tt(e) {
  if (!w(e) || j(e) != Oa)
    return !1;
  var t = Pe(e);
  if (t === null)
    return !0;
  var r = Pa.call(t, "constructor") && t.constructor;
  return typeof r == "function" && r instanceof r && _t.call(r) == Sa;
}
function Ea() {
  this.__data__ = new S(), this.size = 0;
}
function Ca(e) {
  var t = this.__data__, r = t.delete(e);
  return this.size = t.size, r;
}
function Ia(e) {
  return this.__data__.get(e);
}
function ja(e) {
  return this.__data__.has(e);
}
var xa = 200;
function Ma(e, t) {
  var r = this.__data__;
  if (r instanceof S) {
    var n = r.__data__;
    if (!K || n.length < xa - 1)
      return n.push([e, t]), this.size = ++r.size, this;
    r = this.__data__ = new E(n);
  }
  return r.set(e, t), this.size = r.size, this;
}
function A(e) {
  var t = this.__data__ = new S(e);
  this.size = t.size;
}
A.prototype.clear = Ea;
A.prototype.delete = Ca;
A.prototype.get = Ia;
A.prototype.has = ja;
A.prototype.set = Ma;
function La(e, t) {
  return e && R(t, Z(t), e);
}
function Fa(e, t) {
  return e && R(t, D(t), e);
}
var $t = typeof exports == "object" && exports && !exports.nodeType && exports, Ue = $t && typeof module == "object" && module && !module.nodeType && module, Ra = Ue && Ue.exports === $t, Be = Ra ? m.Buffer : void 0, He = Be ? Be.allocUnsafe : void 0;
function At(e, t) {
  if (t)
    return e.slice();
  var r = e.length, n = He ? He(r) : new e.constructor(r);
  return e.copy(n), n;
}
function Da(e, t) {
  for (var r = -1, n = e == null ? 0 : e.length, a = 0, i = []; ++r < n; ) {
    var o = e[r];
    t(o, r, e) && (i[a++] = o);
  }
  return i;
}
function Ot() {
  return [];
}
var Na = Object.prototype, Ga = Na.propertyIsEnumerable, Ke = Object.getOwnPropertySymbols, Ua = Ke ? function(e) {
  return e == null ? [] : (e = Object(e), Da(Ke(e), function(t) {
    return Ga.call(e, t);
  }));
} : Ot;
const Se = Ua;
function Ba(e, t) {
  return R(e, Se(e), t);
}
var Ha = Object.getOwnPropertySymbols, Ka = Ha ? function(e) {
  for (var t = []; e; )
    vt(t, Se(e)), e = Pe(e);
  return t;
} : Ot;
const wt = Ka;
function Wa(e, t) {
  return R(e, wt(e), t);
}
function mt(e, t, r) {
  var n = t(e);
  return $(e) ? n : vt(n, r(e));
}
function ce(e) {
  return mt(e, Z, Se);
}
function za(e) {
  return mt(e, D, wt);
}
var Za = M(m, "DataView");
const le = Za;
var Xa = M(m, "Promise");
const pe = Xa;
var qa = M(m, "Set");
const ge = qa;
var We = "[object Map]", Ya = "[object Object]", ze = "[object Promise]", Ze = "[object Set]", Xe = "[object WeakMap]", qe = "[object DataView]", Ja = x(le), Qa = x(K), Va = x(pe), ka = x(ge), ei = x(ue), C = j;
(le && C(new le(new ArrayBuffer(1))) != qe || K && C(new K()) != We || pe && C(pe.resolve()) != ze || ge && C(new ge()) != Ze || ue && C(new ue()) != Xe) && (C = function(e) {
  var t = j(e), r = t == Ya ? e.constructor : void 0, n = r ? x(r) : "";
  if (n)
    switch (n) {
      case Ja:
        return qe;
      case Qa:
        return We;
      case Va:
        return ze;
      case ka:
        return Ze;
      case ei:
        return Xe;
    }
  return t;
});
const W = C;
var ti = Object.prototype, ri = ti.hasOwnProperty;
function ni(e) {
  var t = e.length, r = new e.constructor(t);
  return t && typeof e[0] == "string" && ri.call(e, "index") && (r.index = e.index, r.input = e.input), r;
}
var ai = m.Uint8Array;
const V = ai;
function Ee(e) {
  var t = new e.constructor(e.byteLength);
  return new V(t).set(new V(e)), t;
}
function ii(e, t) {
  var r = t ? Ee(e.buffer) : e.buffer;
  return new e.constructor(r, e.byteOffset, e.byteLength);
}
var oi = /\w*$/;
function si(e) {
  var t = new e.constructor(e.source, oi.exec(e));
  return t.lastIndex = e.lastIndex, t;
}
var Ye = O ? O.prototype : void 0, Je = Ye ? Ye.valueOf : void 0;
function fi(e) {
  return Je ? Object(Je.call(e)) : {};
}
function Pt(e, t) {
  var r = t ? Ee(e.buffer) : e.buffer;
  return new e.constructor(r, e.byteOffset, e.length);
}
var ui = "[object Boolean]", ci = "[object Date]", li = "[object Map]", pi = "[object Number]", gi = "[object RegExp]", di = "[object Set]", hi = "[object String]", yi = "[object Symbol]", bi = "[object ArrayBuffer]", vi = "[object DataView]", _i = "[object Float32Array]", Ti = "[object Float64Array]", $i = "[object Int8Array]", Ai = "[object Int16Array]", Oi = "[object Int32Array]", wi = "[object Uint8Array]", mi = "[object Uint8ClampedArray]", Pi = "[object Uint16Array]", Si = "[object Uint32Array]";
function Ei(e, t, r) {
  var n = e.constructor;
  switch (t) {
    case bi:
      return Ee(e);
    case ui:
    case ci:
      return new n(+e);
    case vi:
      return ii(e, r);
    case _i:
    case Ti:
    case $i:
    case Ai:
    case Oi:
    case wi:
    case mi:
    case Pi:
    case Si:
      return Pt(e, r);
    case li:
      return new n();
    case pi:
    case hi:
      return new n(e);
    case gi:
      return si(e);
    case di:
      return new n();
    case yi:
      return fi(e);
  }
}
function St(e) {
  return typeof e.constructor == "function" && !$e(e) ? _r(Pe(e)) : {};
}
var Ci = "[object Map]";
function Ii(e) {
  return w(e) && W(e) == Ci;
}
var Qe = F && F.isMap, ji = Qe ? Ae(Qe) : Ii;
const xi = ji;
var Mi = "[object Set]";
function Li(e) {
  return w(e) && W(e) == Mi;
}
var Ve = F && F.isSet, Fi = Ve ? Ae(Ve) : Li;
const Ri = Fi;
var Di = 1, Ni = 2, Gi = 4, Et = "[object Arguments]", Ui = "[object Array]", Bi = "[object Boolean]", Hi = "[object Date]", Ki = "[object Error]", Ct = "[object Function]", Wi = "[object GeneratorFunction]", zi = "[object Map]", Zi = "[object Number]", It = "[object Object]", Xi = "[object RegExp]", qi = "[object Set]", Yi = "[object String]", Ji = "[object Symbol]", Qi = "[object WeakMap]", Vi = "[object ArrayBuffer]", ki = "[object DataView]", eo = "[object Float32Array]", to = "[object Float64Array]", ro = "[object Int8Array]", no = "[object Int16Array]", ao = "[object Int32Array]", io = "[object Uint8Array]", oo = "[object Uint8ClampedArray]", so = "[object Uint16Array]", fo = "[object Uint32Array]", d = {};
d[Et] = d[Ui] = d[Vi] = d[ki] = d[Bi] = d[Hi] = d[eo] = d[to] = d[ro] = d[no] = d[ao] = d[zi] = d[Zi] = d[It] = d[Xi] = d[qi] = d[Yi] = d[Ji] = d[io] = d[oo] = d[so] = d[fo] = !0;
d[Ki] = d[Ct] = d[Qi] = !1;
function Y(e, t, r, n, a, i) {
  var o, s = t & Di, f = t & Ni, p = t & Gi;
  if (r && (o = a ? r(e, n, a, i) : r(e)), o !== void 0)
    return o;
  if (!P(e))
    return e;
  var u = $(e);
  if (u) {
    if (o = ni(e), !s)
      return ft(e, o);
  } else {
    var l = W(e), c = l == Ct || l == Wi;
    if (B(e))
      return At(e, s);
    if (l == It || l == Et || c && !a) {
      if (o = f || c ? {} : St(e), !s)
        return f ? Wa(e, Fa(o, e)) : Ba(e, La(o, e));
    } else {
      if (!d[l])
        return a ? e : {};
      o = Ei(e, l, s);
    }
  }
  i || (i = new A());
  var y = i.get(e);
  if (y)
    return y;
  i.set(e, o), Ri(e) ? e.forEach(function(v) {
    o.add(Y(v, t, r, v, e, i));
  }) : xi(e) && e.forEach(function(v, g) {
    o.set(g, Y(v, t, r, g, e, i));
  });
  var b = p ? f ? za : ce : f ? D : Z, _ = u ? void 0 : b(e);
  return jr(_ || e, function(v, g) {
    _ && (g = v, v = e[g]), ut(o, g, Y(v, t, r, g, e, i));
  }), o;
}
var uo = 1, co = 4;
function lo(e) {
  return Y(e, uo | co);
}
var po = "__lodash_hash_undefined__";
function go(e) {
  return this.__data__.set(e, po), this;
}
function ho(e) {
  return this.__data__.has(e);
}
function k(e) {
  var t = -1, r = e == null ? 0 : e.length;
  for (this.__data__ = new E(); ++t < r; )
    this.add(e[t]);
}
k.prototype.add = k.prototype.push = go;
k.prototype.has = ho;
function yo(e, t) {
  for (var r = -1, n = e == null ? 0 : e.length; ++r < n; )
    if (t(e[r], r, e))
      return !0;
  return !1;
}
function bo(e, t) {
  return e.has(t);
}
var vo = 1, _o = 2;
function jt(e, t, r, n, a, i) {
  var o = r & vo, s = e.length, f = t.length;
  if (s != f && !(o && f > s))
    return !1;
  var p = i.get(e), u = i.get(t);
  if (p && u)
    return p == t && u == e;
  var l = -1, c = !0, y = r & _o ? new k() : void 0;
  for (i.set(e, t), i.set(t, e); ++l < s; ) {
    var b = e[l], _ = t[l];
    if (n)
      var v = o ? n(_, b, l, t, e, i) : n(b, _, l, e, t, i);
    if (v !== void 0) {
      if (v)
        continue;
      c = !1;
      break;
    }
    if (y) {
      if (!yo(t, function(g, T) {
        if (!bo(y, T) && (b === g || a(b, g, r, n, i)))
          return y.push(T);
      })) {
        c = !1;
        break;
      }
    } else if (!(b === _ || a(b, _, r, n, i))) {
      c = !1;
      break;
    }
  }
  return i.delete(e), i.delete(t), c;
}
function To(e) {
  var t = -1, r = Array(e.size);
  return e.forEach(function(n, a) {
    r[++t] = [a, n];
  }), r;
}
function $o(e) {
  var t = -1, r = Array(e.size);
  return e.forEach(function(n) {
    r[++t] = n;
  }), r;
}
var Ao = 1, Oo = 2, wo = "[object Boolean]", mo = "[object Date]", Po = "[object Error]", So = "[object Map]", Eo = "[object Number]", Co = "[object RegExp]", Io = "[object Set]", jo = "[object String]", xo = "[object Symbol]", Mo = "[object ArrayBuffer]", Lo = "[object DataView]", ke = O ? O.prototype : void 0, fe = ke ? ke.valueOf : void 0;
function Fo(e, t, r, n, a, i, o) {
  switch (r) {
    case Lo:
      if (e.byteLength != t.byteLength || e.byteOffset != t.byteOffset)
        return !1;
      e = e.buffer, t = t.buffer;
    case Mo:
      return !(e.byteLength != t.byteLength || !i(new V(e), new V(t)));
    case wo:
    case mo:
    case Eo:
      return z(+e, +t);
    case Po:
      return e.name == t.name && e.message == t.message;
    case Co:
    case jo:
      return e == t + "";
    case So:
      var s = To;
    case Io:
      var f = n & Ao;
      if (s || (s = $o), e.size != t.size && !f)
        return !1;
      var p = o.get(e);
      if (p)
        return p == t;
      n |= Oo, o.set(e, t);
      var u = jt(s(e), s(t), n, a, i, o);
      return o.delete(e), u;
    case xo:
      if (fe)
        return fe.call(e) == fe.call(t);
  }
  return !1;
}
var Ro = 1, Do = Object.prototype, No = Do.hasOwnProperty;
function Go(e, t, r, n, a, i) {
  var o = r & Ro, s = ce(e), f = s.length, p = ce(t), u = p.length;
  if (f != u && !o)
    return !1;
  for (var l = f; l--; ) {
    var c = s[l];
    if (!(o ? c in t : No.call(t, c)))
      return !1;
  }
  var y = i.get(e), b = i.get(t);
  if (y && b)
    return y == t && b == e;
  var _ = !0;
  i.set(e, t), i.set(t, e);
  for (var v = o; ++l < f; ) {
    c = s[l];
    var g = e[c], T = t[c];
    if (n)
      var X = o ? n(T, g, c, t, e, i) : n(g, T, c, e, t, i);
    if (!(X === void 0 ? g === T || a(g, T, r, n, i) : X)) {
      _ = !1;
      break;
    }
    v || (v = c == "constructor");
  }
  if (_ && !v) {
    var N = e.constructor, L = t.constructor;
    N != L && "constructor" in e && "constructor" in t && !(typeof N == "function" && N instanceof N && typeof L == "function" && L instanceof L) && (_ = !1);
  }
  return i.delete(e), i.delete(t), _;
}
var Uo = 1, et = "[object Arguments]", tt = "[object Array]", q = "[object Object]", Bo = Object.prototype, rt = Bo.hasOwnProperty;
function Ho(e, t, r, n, a, i) {
  var o = $(e), s = $(t), f = o ? tt : W(e), p = s ? tt : W(t);
  f = f == et ? q : f, p = p == et ? q : p;
  var u = f == q, l = p == q, c = f == p;
  if (c && B(e)) {
    if (!B(t))
      return !1;
    o = !0, u = !1;
  }
  if (c && !u)
    return i || (i = new A()), o || Oe(e) ? jt(e, t, r, n, a, i) : Fo(e, t, f, r, n, a, i);
  if (!(r & Uo)) {
    var y = u && rt.call(e, "__wrapped__"), b = l && rt.call(t, "__wrapped__");
    if (y || b) {
      var _ = y ? e.value() : e, v = b ? t.value() : t;
      return i || (i = new A()), a(_, v, r, n, i);
    }
  }
  return c ? (i || (i = new A()), Go(e, t, r, n, a, i)) : !1;
}
function Ce(e, t, r, n, a) {
  return e === t ? !0 : e == null || t == null || !w(e) && !w(t) ? e !== e && t !== t : Ho(e, t, r, n, Ce, a);
}
var Ko = 1, Wo = 2;
function zo(e, t, r, n) {
  var a = r.length, i = a, o = !n;
  if (e == null)
    return !i;
  for (e = Object(e); a--; ) {
    var s = r[a];
    if (o && s[2] ? s[1] !== e[s[0]] : !(s[0] in e))
      return !1;
  }
  for (; ++a < i; ) {
    s = r[a];
    var f = s[0], p = e[f], u = s[1];
    if (o && s[2]) {
      if (p === void 0 && !(f in e))
        return !1;
    } else {
      var l = new A();
      if (n)
        var c = n(p, u, f, e, t, l);
      if (!(c === void 0 ? Ce(u, p, Ko | Wo, n, l) : c))
        return !1;
    }
  }
  return !0;
}
function xt(e) {
  return e === e && !P(e);
}
function Zo(e) {
  for (var t = Z(e), r = t.length; r--; ) {
    var n = t[r], a = e[n];
    t[r] = [n, a, xt(a)];
  }
  return t;
}
function Mt(e, t) {
  return function(r) {
    return r == null ? !1 : r[e] === t && (t !== void 0 || e in Object(r));
  };
}
function Xo(e) {
  var t = Zo(e);
  return t.length == 1 && t[0][2] ? Mt(t[0][0], t[0][1]) : function(r) {
    return r === e || zo(r, e, t);
  };
}
function qo(e, t) {
  return e != null && t in Object(e);
}
function Yo(e, t, r) {
  t = yt(t, e);
  for (var n = -1, a = t.length, i = !1; ++n < a; ) {
    var o = ie(t[n]);
    if (!(i = e != null && r(e, o)))
      break;
    e = e[o];
  }
  return i || ++n != a ? i : (a = e == null ? 0 : e.length, !!a && Te(a) && _e(o, a) && ($(e) || Q(e)));
}
function Jo(e, t) {
  return e != null && Yo(e, t, qo);
}
var Qo = 1, Vo = 2;
function ko(e, t) {
  return we(e) && xt(t) ? Mt(ie(e), t) : function(r) {
    var n = $a(r, e);
    return n === void 0 && n === t ? Jo(r, e) : Ce(t, n, Qo | Vo);
  };
}
function es(e) {
  return function(t) {
    return t == null ? void 0 : t[e];
  };
}
function ts(e) {
  return function(t) {
    return bt(t, e);
  };
}
function rs(e) {
  return we(e) ? es(ie(e)) : ts(e);
}
function ns(e) {
  return typeof e == "function" ? e : e == null ? be : typeof e == "object" ? $(e) ? ko(e[0], e[1]) : Xo(e) : rs(e);
}
function as(e) {
  return function(t, r, n) {
    for (var a = -1, i = Object(t), o = n(t), s = o.length; s--; ) {
      var f = o[e ? s : ++a];
      if (r(i[f], f, i) === !1)
        break;
    }
    return t;
  };
}
var is = as();
const Lt = is;
function os(e, t) {
  return e && Lt(e, t, Z);
}
function de(e, t, r) {
  (r !== void 0 && !z(e[t], r) || r === void 0 && !(t in e)) && te(e, t, r);
}
function ss(e) {
  return w(e) && re(e);
}
function he(e, t) {
  if (!(t === "constructor" && typeof e[t] == "function") && t != "__proto__")
    return e[t];
}
function fs(e) {
  return R(e, D(e));
}
function us(e, t, r, n, a, i, o) {
  var s = he(e, r), f = he(t, r), p = o.get(f);
  if (p) {
    de(e, r, p);
    return;
  }
  var u = i ? i(s, f, r + "", e, t, o) : void 0, l = u === void 0;
  if (l) {
    var c = $(f), y = !c && B(f), b = !c && !y && Oe(f);
    u = f, c || y || b ? $(s) ? u = s : ss(s) ? u = ft(s) : y ? (l = !1, u = At(f, !0)) : b ? (l = !1, u = Pt(f, !0)) : u = [] : Tt(f) || Q(f) ? (u = s, Q(s) ? u = fs(s) : (!P(s) || ve(s)) && (u = St(f))) : l = !1;
  }
  l && (o.set(f, u), a(u, f, n, i, o), o.delete(f)), de(e, r, u);
}
function Ft(e, t, r, n, a) {
  e !== t && Lt(t, function(i, o) {
    if (a || (a = new A()), P(i))
      us(e, t, o, r, Ft, n, a);
    else {
      var s = n ? n(he(e, o), i, o + "", e, t, a) : void 0;
      s === void 0 && (s = i), de(e, o, s);
    }
  }, D);
}
var cs = ct(function(e, t, r, n) {
  Ft(e, t, r, n);
});
const ls = cs;
function ps(e, t) {
  var r = {};
  return t = ns(t), os(e, function(n, a, i) {
    te(r, t(n, a, i), n);
  }), r;
}
var ee = function() {
  return ee = Object.assign || function(t) {
    for (var r, n = 1, a = arguments.length; n < a; n++) {
      r = arguments[n];
      for (var i in r)
        Object.prototype.hasOwnProperty.call(r, i) && (t[i] = r[i]);
    }
    return t;
  }, ee.apply(this, arguments);
};
function gs(e) {
  return e.toLowerCase();
}
var ds = [/([a-z0-9])([A-Z])/g, /([A-Z])([A-Z][a-z])/g], hs = /[^A-Z0-9]+/gi;
function ys(e, t) {
  t === void 0 && (t = {});
  for (var r = t.splitRegexp, n = r === void 0 ? ds : r, a = t.stripRegexp, i = a === void 0 ? hs : a, o = t.transform, s = o === void 0 ? gs : o, f = t.delimiter, p = f === void 0 ? " " : f, u = nt(nt(e, n, "$1\0$2"), i, "\0"), l = 0, c = u.length; u.charAt(l) === "\0"; )
    l++;
  for (; u.charAt(c - 1) === "\0"; )
    c--;
  return u.slice(l, c).split("\0").map(s).join(p);
}
function nt(e, t, r) {
  return t instanceof RegExp ? e.replace(t, r) : t.reduce(function(n, a) {
    return n.replace(a, r);
  }, e);
}
function Rt(e, t) {
  var r = e.charAt(0), n = e.substr(1).toLowerCase();
  return t > 0 && r >= "0" && r <= "9" ? "_" + r + n : "" + r.toUpperCase() + n;
}
function bs(e, t) {
  return t === void 0 && (t = {}), ys(e, ee({ delimiter: "", transform: Rt }, t));
}
function vs(e, t) {
  return t === 0 ? e.toLowerCase() : Rt(e, t);
}
function _s(e, t) {
  return t === void 0 && (t = {}), bs(e, ee({ transform: vs }, t));
}
function Ts(e, t) {
  const r = /* @__PURE__ */ Object.create(null), n = e.split(",");
  for (let a = 0; a < n.length; a++)
    r[n[a]] = !0;
  return t ? (a) => !!r[a.toLowerCase()] : (a) => !!r[a];
}
const $s = /* @__PURE__ */ Ts(
  "String,Number,Boolean,Function,Symbol,BigInt"
);
function As(e) {
  const t = e && e.toString().match(/^\s*function (\w+)/);
  return t ? t[1] : e === null ? "null" : "";
}
function Os(e, t) {
  let r;
  const n = As(t);
  if ($s(n)) {
    const a = typeof e;
    r = a === n.toLowerCase(), !r && a === "object" && (r = e instanceof t);
  } else
    n === "Object" ? r = P(e) : n === "Array" ? r = Array.isArray(e) : n === "null" ? r = e === null : r = e instanceof t;
  return {
    valid: r,
    expectedType: n
  };
}
function ws({
  prop: e,
  type: t,
  validator: r
}) {
  if (![void 0, null].includes(e) && t) {
    let n = !1;
    const a = Array.isArray(t) ? t : [t], i = [];
    for (let o = 0; o < a.length && !n; o++) {
      const { valid: s, expectedType: f } = Os(e, a[o]);
      i.push(f || ""), n = s;
    }
    if (!n)
      throw new Error(`Invalid prop: type check failed, expecting [${i.join(", ")}], receiving: ${e}`);
  }
  if (r && !r(e))
    throw new Error(`Invalid prop: validator check failed, receiving: ${e}`);
}
function ms(e, {
  mergeObject: t,
  mergeFunction: r
}) {
  const n = [];
  for (let i = e.length - 1; i >= 0; i--)
    n.push(e[i]);
  const a = r ? (i, o) => typeof i == "function" && typeof o == "function" ? r(o, i) : void 0 : void 0;
  return t === "deep" ? ls(...n, a) : Dn(...n, a);
}
function Ps(e, {
  mergeFunction: t
}) {
  return e.reduce(t, () => {
  });
}
function Ss(e, t = {}) {
  const {
    type: r,
    default: n,
    defaultIsDynamic: a = !1,
    required: i = !1,
    validator: o,
    camelizeObjectKeys: s = !1,
    mergeObjectApplyOnlyToDefault: f = !1,
    mergeFunctionApplyOnlyToDefault: p = !0
  } = t;
  let {
    mergeObject: u = "deep",
    mergeFunction: l = !1
  } = t;
  const c = [];
  let y, b = !1, _ = !1;
  const v = (g) => {
    if (g !== void 0) {
      ws({ type: r, prop: g, validator: o });
      const T = Tt(g), X = typeof g == "function";
      return b = T, _ = X, T ? (g = lo(g), s ? ps(g, (N, L) => _s(L, {
        stripRegexp: /-/g
        // Filter only short horizontal lines for kebab-case conversion to camelCase
      })) : g) : g;
    }
  };
  for (const g of e)
    c.push(v(g));
  if (!a)
    c.push(v(n));
  else if (typeof n != "function")
    throw new TypeError(`Invalid option: config.default should be Function when config.defaultIsDynamic enabled, receiving: ${n}`);
  b || (u = !1, _ || (l = !1));
  for (let g = 0; g < c.length; g++) {
    const T = c[g];
    if (T !== void 0) {
      g === c.length - 1 ? y = T : u ? y = ms(f ? [T, n] : c, {
        mergeObject: u,
        mergeFunction: l
      }) : l ? y = Ps(p ? [T, n] : c, {
        mergeFunction: l
      }) : y = T;
      break;
    }
  }
  if (i && [void 0, null].includes(y))
    throw new Error("Missing required prop");
  return a ? Ss(
    e,
    {
      ...t,
      default: n(y),
      defaultIsDynamic: !1
    }
  ) : y;
}
function Es(e, t) {
  return Object.keys(e).filter((r) => t(r)).reduce((r, n) => (r[n] = e[n], r), {});
}
function Is(e) {
  return Es(e, (t) => !t.startsWith("hook:"));
}
function js(e) {
  if (Object.getOwnPropertyNames(e || {}).length) {
    const t = this.$emit;
    this.$emit = function(...r) {
      var i;
      t.apply(this, r);
      const [n, ...a] = r;
      (i = e[n]) == null || i.apply(this, a);
    };
  }
}
function at(e) {
  const t = Array.from(e);
  return t[0] = t[0].toUpperCase(), t.unshift("n"), t.unshift("o"), t.join("");
}
function xs(e, t = []) {
  const r = {
    props: {},
    attrs: {},
    listeners: {},
    hooks: {},
    slots: {}
  }, n = Array.isArray(t) ? t : Object.keys(t);
  for (const a in e)
    if (a.startsWith("@")) {
      const i = a.substring(1);
      Dt ? i.startsWith("vnode") ? r.hooks[at(i)] = e[a] : r.listeners[at(i)] = e[a] : i.startsWith("hook:") ? r.hooks[i] = e[a] : r.listeners[i] = e[a];
    } else if (a.startsWith("#")) {
      const i = a.substring(1);
      i && (r.slots[i] = e[a]);
    } else
      n.includes(a) ? r.props[a] = e[a] : r.attrs[a] = e[a];
  return r;
}
export {
  Ss as conclude,
  Is as getLocalListeners,
  js as listenGlobalHooks,
  xs as resolveConfig
};
